builder/amazon/common/step_run_source_instance.go

```diff
@@ -239,3 +237,22 @@ func (s *StepRunSourceInstance) Cleanup(state multistep.StateBag) {
 		}
 	}
 }
+
+func waitUntilInstanceReady(ec2conn *ec2.EC2, instanceID, instanceType string) error {
+	statusOKMap := make(map[string]struct{})
+	statusOKMap["c5"] = struct{}{}
```

Comment:
I think we should have a comment explaining _why_ this convoluted extra code for c5.  something like // c5 instances need us to use the time-sucking WaitUntilStatusOK feature instead of just waiting for SSH to become available

Reply:
Pretty sure the new m5 instance type is also backed by the new hypervisor. Ought to be easy to test if the issue is limited to c5, or if a more general approach is needed.